### PR TITLE
Improve link_package speed

### PR DESCRIPTION
**Pull request by mattsoulanille**
Improve `link_package` speed by only building the dependencies that are required and by building them all in a single `bazel build` command.

Remove the `link_package_core` package, which is no longer needed now that `link_package` can build just the required dependencies.

To see the logs from the Cloud Build CI, please join either our [discussion](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs) or [announcement](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs-announce) mailing list.

---
This change is [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/6376)

### CODE REVIEW

File: BAZEL_MIGRATION.md [hunk 1]
```diff
@@ -364,31 +364,12 @@ As a core featue of its design, Bazel places outputs in a different directory th
 Instead, we maintain a `link-package` pseudopackage where we copy the Bazel outputs. This package allows for correct Node module resolution between Bazel outputs because it has its own `node_modules` folder. This package will never be published and will be removed once the migration is complete.
 
 #### Add the Package to `link-package`
-Add your package to the `devDependencies` of the `link-package`'s `package.json`. The package path should be similar to the one below and is based off of the `pkg_npm` rule's name.
+Add your package to the the `PACKAGES` list in the `build_deps.ts` script in `link-package`. For a package with npm name `@tensorflow/tfjs-foo`, the package's directory in the monorepo and the value to add to `PACKAGES` should both be `tfjs-foo`. The name of the package's `pkg_npm` target should be `tfjs-foo_pkg`.
```
Comment (jinjingforever) [inline BAZEL_MIGRATION.md:367]: nit: remove "the" before "PACKAGES list".

**Review by jinjingforever — APPROVED** [review]

File: BAZEL_MIGRATION.md (hunk 1 above)
Comment (mattsoulanille) [reply BAZEL_MIGRATION.md]: Done

**Review by pyu10055 — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/6376)* status: :shipit: complete! 2 of 1 approvals obtained (waiting on @mattsoulanille)

___
*[`BAZEL_MIGRATION.md` line 381 at r12](https://reviewable.io/reviews/tensorflow/tfjs/6376#-N1kAgIe2XzP1bM7dSsZ:-N1kAgIe2XzP1bM7dSs_:bfssvwr) ([raw file](https://github.com/tensorflow/tfjs/blob/cc07f49eb347ed56c28a7dee6f3bb39f47f3abd4/BAZEL_MIGRATION.md#L381)):*
> ```Markdown
> "devDependencies": {
>   "@tensorflow/tfjs-core": "link:../link-package/node_modules/@tensorflow/tfjs-core",
>   "@tensorflow/your-package": "link:../link-package/node_modules/@tensorflow/your-package",
> ```

should `your-package` be changed to `tfjs-foo` to match the context?
___
*[`BAZEL_MIGRATION.md` line 391 at r12](https://reviewable.io/reviews/tensorflow/tfjs/6376#-N1kAv0WCE0hLojdYsHr:-N1kAv0WCE0hLojdYsHs:b-1ci4o4) ([raw file](https://github.com/tensorflow/tfjs/blob/cc07f49eb347ed56c28a7dee6f3bb39f47f3abd4/BAZEL_MIGRATION.md#L391)):*
> ```Markdown
> ```json
> "scripts": {
>   "build-link-package": "cd ../link-package && yarn build-deps-for tfjs-backend-cpu tfjs-foo",
> ```

can you explain the relationship of `tfjs-backend-cpu` and `tfjs-foo`?
___
*[`link-package/build_deps.ts` line 43 at r12](https://reviewable.io/reviews/tensorflow/tfjs/6376#-N1kBMkLAj9nLNZ8wYjq:-N1kBMkM3EcGF2HPafQP:ba1n6la) ([raw file](https://github.com/tensorflow/tfjs/blob/cc07f49eb347ed56c28a7dee6f3bb39f47f3abd4/link-package/build_deps.ts#L43)):*
> ```XML
> 
> /**
>  * Build bazel dependencies for the package specified by tfjs_package.
> ```

should this be called `tfjs_packages`? seems it allows multiple arguments

**Review by mattsoulanille — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/6376)* status: :shipit: complete! 2 of 1 approvals obtained (waiting on @pyu10055)

___
*[`BAZEL_MIGRATION.md` line 381 at r12](https://reviewable.io/reviews/tensorflow/tfjs/6376#-N1kAgIe2XzP1bM7dSsZ:-N1kG6yW3ZNLvblAR2uY:buc623c) ([raw file](https://github.com/tensorflow/tfjs/blob/cc07f49eb347ed56c28a7dee6f3bb39f47f3abd4/BAZEL_MIGRATION.md#L381)):*
<details><summary><i>Previously, pyu10055 (Ping Yu) wrote…</i></summary><blockquote>

should `your-package` be changed to `tfjs-foo` to match the context?
</blockquote></details>

Yes, thanks! Fixed.
___
*[`BAZEL_MIGRATION.md` line 391 at r12](https://reviewable.io/reviews/tensorflow/tfjs/6376#-N1kAv0WCE0hLojdYsHr:-N1kGCbkBkQ4g4ZBc5nb:baqh014) ([raw file](https://github.com/tensorflow/tfjs/blob/cc07f49eb347ed56c28a7dee6f3bb39f47f3abd4/BAZEL_MIGRATION.md#L391)):*
<details><summary><i>Previously, pyu10055 (Ping Yu) wrote…</i></summary><blockquote>

can you explain the relationship of `tfjs-backend-cpu` and `tfjs-foo`?
</blockquote></details>

They are both dependencies (possibly transitive dependencies) of the downstream package. I changed `tfjs-backend-cpu` to `tfjs-some-other-dependency` to make it more clear that you don't specifically need to list `tfjs-backend-cpu`. You just need the dependencies of the downstream package.
___
*[`link-package/build_deps.ts` line 43 at r12](https://reviewable.io/reviews/tensorflow/tfjs/6376#-N1kBMkLAj9nLNZ8wYjq:-N1kGnr2-L2LMM5SNzI6:bfnf2h1) ([raw file](https://github.com/tensorflow/tfjs/blob/cc07f49eb347ed56c28a7dee6f3bb39f47f3abd4/link-package/build_deps.ts#L43)):*
<details><summary><i>Previously, pyu10055 (Ping Yu) wrote…</i></summary><blockquote>

should this be called `tfjs_packages`? seems it allows multiple arguments
</blockquote></details>

It's a repeated arg, so I think the argument should be singular, since the help message prints it like this:
```
usage: build_deps.ts [-h] [--all] [tfjs_package ...]

positional arguments:
  tfjs_package  tfjs package to build dependencies for

optional arguments:
  -h, --help    show this help message and exit
  --all         Build all packages
Done in 0.81s.
```
implying it can be repeated. I updated the doccomment, though.

**Pull request merged**